docs/storybook/src/AppUiStory.tsx

```diff
@@ -14,62 +15,138 @@ import {
 import { createFrontstageProvider } from "./Utils";
+import { DemoIModel, useDemoIModel } from "../.storybook/addons/DemoIModel";
 
 export interface AppUiStoryProps {
+  appBackstage?: React.ReactNode;
+  children?: React.ReactNode;
   onInitialize?: () => Promise<void>;
   itemProviders?: UiItemsProvider[];
   layout?: "fullscreen";
   frontstageProviders?: FrontstageProvider[] | (() => FrontstageProvider[]);
+  demoIModel?: boolean;
 }
 
 export function AppUiStory(props: AppUiStoryProps) {
+  const demoIModel = useStoryDemoIModel(props);
   const [initialized, setInitialized] = React.useState(false);
+
   React.useEffect(() => {
-    void (async function () {
+    let ignore = false;
+
+    const startup = async () => {
       await IModelApp.startup({
         toolAdmin: new FrameworkToolAdmin(),
+        hubAccess: new FrontendIModelsAccess(
+          new IModelsClient({
+            api: {
+              baseUrl: "https://api.bentley.com/imodels",
+            },
+          })
+        ),
+        authorizationClient: new DemoAuthClient(),
       });
       await UiFramework.initialize(undefined);
+
+      BentleyCloudRpcManager.initializeClient(
+        {
+          info: {
+            title: "visualization",
+            version: "v4.0",
+          },
+          pathPrefix: "https://api.bentley.com/imodeljs",
+        },
+        [
+          IModelReadRpcInterface,
+          IModelTileRpcInterface,
+          SnapshotIModelRpcInterface,
+        ]
+      );
+      demoIModel && (await openIModel(demoIModel));
       await props.onInitialize?.();
 
       const frontstageProviders = getFrontstageProviders(
         props.frontstageProviders
-      ) ?? [createFrontstageProvider()];
+      );
       for (const provider of frontstageProviders) {
         UiFramework.frontstages.addFrontstageProvider(provider);
       }
-
       for (const provider of props.itemProviders ?? []) {
         UiItemsManager.register(provider);
       }
-      const defaultFrontstage = frontstageProviders[0];
-      defaultFrontstage &&
-        (await UiFramework.frontstages.setActiveFrontstage(
-          defaultFrontstage.id
-        ));
+
+      if (ignore) return;
       setInitialized(true);
-    })();
-    return () => {
-      void UiFramework.frontstages.setActiveFrontstageDef(undefined);
+    };
+
+    const shutdown = async () => {
+      await UiFramework.getIModelConnection()?.close();
+      await UiFramework.frontstages.setActiveFrontstageDef(undefined);
       UiFramework.frontstages.clearFrontstageProviders();
       for (const provider of props.itemProviders ?? []) {
         UiItemsManager.unregister(provider.id);
       }
+      UiFramework.terminate();
+      await IModelApp.shutdown();
     };
-  }, [props]);
+
+    const cleanup = appInitializer.initialize(startup, shutdown);
+    return () => {
+      ignore = true;
+      setInitialized(false);
+      cleanup();
+    };
+  }, [props, demoIModel]);
   if (!initialized) return null;
```

Comment:
With that change, switching component/iModel can take a certain time, I suggest adding something so we know something is going on.
```suggestion
  if (!initialized) return <ProgressLinear indeterminate labels={["Getting things ready!"]} />;
```

(Will require the import though...)